config/misc.go

```diff
@@ -35,8 +36,8 @@ func GetInstanceID() string {
 	if instance == "" {
 		return ""
 	}
-	instanceArr := strings.Split(instance, "-")
-	return instanceArr[len(instanceArr)-1]
+	re := regexp.MustCompile("-[0-9]+([A-Za-z0-9-]*)$")
```

Comment:
Please add a test case for this function covering all scenarios & using some actual instance_id values:

1. Scenarios that successfully parse `INSTANCE_ID`
2.Scenarios that fail to parse `INSTANCE_ID` and return an empty string

Specifically for the case of `ETCDManager`, I understand that it is important that instance id is a non-empty string, i.e. a number, thus the least we should do is log a warning when this is empty, or even fail to start.